### PR TITLE
Issue #3430: Temporal Infinity Values

### PR DESCRIPTION
**Pull request by hawkfish**
* Add temporal infinity values.
* Parse temporal infinity values.
* Adjust valid time boundaries to exclude infinities.
* Adding to and subtracting from temporal infinites should produce the same infinity.
* Date part extraction from infinities is NULL.
* Add ICU support for infinities.
* Update temporal functions to handle infinities.
* Disable infinities as range/generate_series bounds

### CODE REVIEW

File: src/function/scalar/operators/add.cpp [hunk 1]
```diff
@@ -44,11 +44,18 @@ interval_t AddOperator::Operation(interval_t left, interval_t right) {
 
 template <>
 date_t AddOperator::Operation(date_t left, int32_t right) {
-	int32_t result;
-	if (!TryAddOperator::Operation(left.days, right, result)) {
+	if (!Value::IsFinite(left)) {
```
Comment (taniabogatsch) [inline src/function/scalar/operators/add.cpp:47]: So this means that `left` is either `infinity` or `-infinity`? In which case any addition would also return `infinity` or `-infinity`?

File: src/function/scalar/operators/add.cpp [hunk 2]
```diff
@@ -62,6 +69,9 @@ timestamp_t AddOperator::Operation(date_t left, dtime_t right) {
 	if (!Timestamp::TryFromDatetime(left, right, result)) {
 		throw OutOfRangeException("Timestamp out of range");
 	}
+	if (!Value::IsFinite(result)) {
```
Comment (taniabogatsch) [inline src/function/scalar/operators/add.cpp:72]: In `date_t AddOperator::Operation(date_t left, int32_t right)` we check for infinities before trying to add stuff, why is this not required here?

**Review by taniabogatsch — COMMENTED** [review]
Great work! :)
I had a look at it, but I think that this review is a bit outside of my capabilities (yet). I added two comments but these are actually just questions.

File: src/function/scalar/operators/add.cpp (hunk 1 above)
Comment (hawkfish) [reply src/function/scalar/operators/add.cpp:47]: Yes. (That is what PG does).

File: src/function/scalar/operators/add.cpp (hunk 2 above)
Comment (hawkfish) [reply src/function/scalar/operators/add.cpp:72]: Good catch! Adding a `TIME` to an infinite `DATE` should produce the same infinity.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. Perhaps we want to add these to the `test_all_types` function so we can ensure these are correctly translated on all different client connectors?

**Comment by hawkfish** [comment]
> Thanks for the PR! Looks good to me. Perhaps we want to add these to the `test_all_types` function so we can ensure these are correctly translated on all different client connectors?

Yeah, that will be fun. Might need some help with ODBC expecteds.

**Review by Mytherin — APPROVED** [review]
Thanks! LGTM now.

**Comment by Mytherin** [comment]
Looks like there was a merge conflict introduced by #3613, could you have a look at fixing that?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**